src/evict/evict_lru.c

```diff
@@ -354,6 +354,40 @@ __evict_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
     return (ret);
 }
 
+/*
+ * __clear_saved_walk_tree --
+ *     Clear saved walk tree maintaining use count.
+ */
+static void
+__clear_saved_walk_tree(WT_SESSION_IMPL *session)
```

Comment:
How about passing in a `WT_DATA_HANDLE**` that is set to the `cache->walk_tree` if it is not NULL. My thought here is to slightly simplify the logic in the main evict loop.

Reply:
"clear" and "set" functions just implement the "smart pointer" pattern. I didn't aim to do any other refactoring here. To keep it simple and do one thing at a time, I wouldn't do it as part of this PR.